extra/checksums/process/process.factor

```diff
@@ -0,0 +1,34 @@
+! Copyright (C) 2016 Alexander Ilin.
+! See http://factorcode.org/license.txt for BSD license.
+USING: accessors checksums checksums.common combinators destructors
+io io.encodings.binary io.launcher kernel math.parser sequences
+strings ;
+IN: checksums.process
+
+TUPLE: checksum-process launch-desc ;
+INSTANCE: checksum-process block-checksum
+C: <checksum-process> checksum-process
+
+TUPLE: process-state < disposable proc result ;
+
+M: checksum-process initialize-checksum-state ( checksum -- checksum-state )
+    launch-desc>> binary <process-stream> process-state new-disposable swap >>proc ;
+
+M: process-state dispose* ( process-state -- )
+    proc>> [ dispose ] when* ;
+
+M: process-state add-checksum-bytes ( process-state bytes -- process-state' )
+    over proc>> stream-write ;
+
+: trim-hash ( str -- str' ) dup " *-" swap start head ;
+
+M: process-state get-checksum ( checksum-state -- value )
+    dup result>> [
+        dup proc>> [
+            [
+                [ out>> dispose ] keep
+                stream-contents >string trim-hash hex-string>bytes
+            ] with-disposal
+        ] [ B{ } clone ] if*
```

Comment:
Could be. Do you think it would be better to throw something?
To my mind, this is the case of a hashing-state before the hashing got started. No hashing - empty result. As good as any. Didn't want to complicate things by adding exceptions.
Other errors may also produce empty result, e.g. if the console app returned nothing:

```
"" hex-string>bytes .
B{ }
```

So, it's a generic empty result, without interrupting the control flow.